docs/how_to/publish.rst

```diff
@@ -257,6 +258,8 @@ Currently, the supported protocols are ``default`` and ``email`` only, and the `
 send notifications.
 The ``default`` message will be used as a fallback message for any protocols.
 
+Also, Regions can also be specified in ``sns`` configuration to indicate where the notification should be sent. If no regions are specified, SNS will default to using all regions in the partition.
```

Comment:
I'm wondering now what behavior we actually want. if no sns regions are specified, do we want to send to **all** regions within the partition or do we want to sent to all regions configured for the used image? Or do we want to make that configurable?

Reply:
i would lean towards only sending notifications for regions that are configured (unless specific sns regions are configured). that feels more consistent with the rest of the impl and is more closely aligned with the use cases.

Reply:
I discussed this with @rthill91 at the beginning, and we agreed it would be best to follow the approach used by `image_region` for consistency. Also, We’ve decided that users can send notifications regardless of the regions where the image has been published/registered and the publish will fail if the region doesn’t have the topic. But let me know if this should be sent within image_regions.

Reply:
I like the idea of sending to all regions if none are specified because it mimics what `regions` does for images. My only real concern with having sns regions effectively inherit regions from images is that it creates another 'hidden' bit of config. At least falling back to all regions means the only 'hidden' config is "use all regions if undefined". That said, I suspect most users are going to want to either a single region, or the regions that match there image publication, so it would likely lead to more duplication

Reply:
Thinking more about and and talking to Jess, I think we should do what we do for images (what Ryan says here). So if no regions configured, send to all regions. So the logic for region handling is consistent beween sns and an image.